template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/zeppelin/ZeppelinS3CloudStorageRoleConfigProvider.java

```diff
@@ -27,10 +27,10 @@
 
     private static final String ZEPPELIN_NOTEBOOK_S3_USER = "zeppelin.notebook.s3.user";
 
-    private static final String ZEPPELIN_DEFAULT_USER = "zeppelin";
-
     private static final String ZEPPELIN_S3_NOTEBOOK_REPO = "org.apache.zeppelin.notebook.repo.S3NotebookRepo";
 
+    private static final Pattern S3A_BUCKET_PATTERN = Pattern.compile(FileSystemType.S3.getProtocol() + "://([\\w.-]*)/(.*)");
```

Comment:
Minor issues:
* I would rather change the `*` in `[\\w.-]*` to a `+` as bucket names obviously cannot be empty. 😃 
* Is the `/` terminating the bucket name mandatory in case of an empty path? If not, the second half of the pattern could become something like `(?:/(.*))?`.

Reply:
Changed `*` to `+` :)
Notebook storage has zeppelin/notebook suffixed hence there cannot be an empty path since it would create notebooks in bucket root location, but still updated since while launching via CLI there is no additional validation.

Reply:
Perfect, thanks!